rust/yarp-sys/build.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+fn main() -> Result<(), io::Error> {
+    rake_compile()?;
+
+    let ruby_build_path = ruby_build_path();
+    let ruby_include_path = ruby_include_path();
+
+    // Tell cargo/rustc that we want to link against `librubyparser.a`.
+    println!("cargo:rustc-link-lib=static=rubyparser");
+
+    // Add `[root]/build/` to the search paths, so it can find `librubyparser.a`.
+    println!(
+        "cargo:rustc-link-search=native={}",
+        ruby_build_path.display()
```

Comment:
if the path is not able to be represented as UTF-8, this is unlikely to behave correctly, so prefer to abort.

```suggestion
        ruby_build_path.to_str().unwrap()
```

Reply:
Interesting. I thought the `println!` in build scripts effectively behaves like sending the string via stdin, and in that case you'd want to whatever encoding the OsStr was in. Why is there a need to ensure a UTF-8 string is used here?

Reply:
`println!` by virtue of taking a `&str` format string requires the fully constructed payload to be UTF-8.

`Path::display` results in a UTF-8 encoded representation of the path, which for paths with invalid UTF-8 contents like `\xFF` or unpaired surrogates on Windows, will not be a lossless transformation (it will include some escaping).

Such paths are unable to be losslessly printed in this way, which means paths containing them won't properly set up the search path.

Rather than delaying this scenario to a more opaque error later in linking, prefer to panic early.

Reply:
Ah ok, that makes sense. I think I read the `Path::display` docs to say that it'd allow non-unicode strings, but didn't catch the lossy part (which certainly makes it sound like it behaves similar to the various `to_string_lossy()` calls in `std`). I'll change. Thanks for the explanation.